File: CS/DataBindingToListExample/Fishes.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;

namespace DataBindingToListExample {

    public class MyFish {
        [DisplayName("ID")]
        public int ID { get; set; }
        [DisplayName("Fish Category")]
        public string Category { get; set; }
        [DisplayName("Fish Common Name")]
        public string CommonName { get; set; }
        [DisplayName("Fish Species Name")]
        public string SpeciesName { get; set; }
        [DisplayName("Hyperlink")]
        public string Hyperlink { get; set; }
        [DisplayName("TimeStamp")]
        public DateTime TimeStamp { get; set; }
        public MyFish(int id, string category, string commonName, string speciesName, string hyperlink, DateTime timeStamp) {
            this.ID = id;
            this.Category = category;
            this.CommonName = commonName;
            this.SpeciesName = speciesName;
            this.Hyperlink = hyperlink;
            this.TimeStamp = timeStamp;
        }
    }
    public class Fish {
        [DisplayName("ID")]
        public int ID { get; set; }
        [DisplayName("Category")]
        public string Category { get; set; }
        [DisplayName("Common Name")]
        public string CommonName { get; set; }
        [DisplayName("Notes")]
        public string Notes { get; set; }
        [DisplayName("SpeciesName")]
        public string SpeciesName
        {
            get
            {
                return this.ScientificClassification.Species;
            }
            set
            {
                this.ScientificClassification.Species = value;
            }
        }
        [DisplayName("Hyperlink")]
        public string Hyperlink
        {
            get
            {
                return this.ScientificClassification.Hyperlink;
            }
            set
            {
                this.ScientificClassification.Hyperlink = value;
            }
        }
        public ScientificClassification ScientificClassification { get; set; }
    }
    public class ScientificClassification {
        [XmlElement("Reference")]
        public string Hyperlink { get; set; }
        public string Kingdom { get; set; }
        public string Phylum { get; set; }
        [XmlElement("Class")]
        [DisplayName("Class")]
        public string _Class { get; set; }
        public string Order { get; set; }
        public string Family { get; set; }
        public string Genus { get; set; }
        public string Species { get; set; }
    }
    public static class MyFishesSource {
        static List<MyFish> data;
        public static System.Collections.ObjectModel.ReadOnlyCollection<MyFish> Data
        {
            get
            {
                Random rnd1 = new Random();
                if (data == null) {
                    List<Fish> fLIst = GetDataSource();
                    data = new List<MyFish>();
                    foreach (Fish f in fLIst) {
                        data.Add(new MyFish(f.ID, f.Category, f.CommonName, f.SpeciesName, f.Hyperlink, DateTime.Now.AddHours(-24 * rnd1.NextDouble())));
                    }
                }
                return data.AsReadOnly();
            }
        }
        static List<Fish> GetDataSource() {
            return DataSourceHelper.GetDataSouresFromXml<Fish>("fishes.xml", "Fishes");
        }
    }
}

File: VB/DataBindingToListExample/Fishes.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Xml.Serialization

Namespace DataBindingToListExample

    Public Class MyFish
        <DisplayName("ID")> _
        Public Property ID() As Integer
        <DisplayName("Fish Category")> _
        Public Property Category() As String
        <DisplayName("Fish Common Name")> _
        Public Property CommonName() As String
        <DisplayName("Fish Species Name")> _
        Public Property SpeciesName() As String
        <DisplayName("Hyperlink")> _
        Public Property Hyperlink() As String
        <DisplayName("TimeStamp")> _
        Public Property TimeStamp() As Date
        Public Sub New(ByVal id As Integer, ByVal category As String, ByVal commonName As String, ByVal speciesName As String, ByVal hyperlink As String, ByVal timeStamp As Date)
            Me.ID = id
            Me.Category = category
            Me.CommonName = commonName
            Me.SpeciesName = speciesName
            Me.Hyperlink = hyperlink
            Me.TimeStamp = timeStamp
        End Sub
    End Class
    Public Class Fish
        <DisplayName("ID")> _
        Public Property ID() As Integer
        <DisplayName("Category")> _
        Public Property Category() As String
        <DisplayName("Common Name")> _
        Public Property CommonName() As String
        <DisplayName("Notes")> _
        Public Property Notes() As String
        <DisplayName("SpeciesName")> _
        Public Property SpeciesName() As String
            Get
                Return Me.ScientificClassification.Species
            End Get
            Set(ByVal value As String)
                Me.ScientificClassification.Species = value
            End Set
        End Property
        <DisplayName("Hyperlink")> _
        Public Property Hyperlink() As String
            Get
                Return Me.ScientificClassification.Hyperlink
            End Get
            Set(ByVal value As String)
                Me.ScientificClassification.Hyperlink = value
            End Set
        End Property
        Public Property ScientificClassification() As ScientificClassification
    End Class
    Public Class ScientificClassification
        <XmlElement("Reference")> _
        Public Property Hyperlink() As String
        Public Property Kingdom() As String
        Public Property Phylum() As String
        <XmlElement("Class"), DisplayName("Class")> _
        Public Property _Class() As String
        Public Property Order() As String
        Public Property Family() As String
        Public Property Genus() As String
        Public Property Species() As String
    End Class
    Public NotInheritable Class MyFishesSource

        Private Sub New()
        End Sub


        Private Shared data_Renamed As List(Of MyFish)
        Public Shared ReadOnly Property Data() As System.Collections.ObjectModel.ReadOnlyCollection(Of MyFish)
            Get
                Dim rnd1 As New Random()
                If data_Renamed Is Nothing Then
                    Dim fLIst As List(Of Fish) = GetDataSource()
                    data_Renamed = New List(Of MyFish)()
                    For Each f As Fish In fLIst
                        data_Renamed.Add(New MyFish(f.ID, f.Category, f.CommonName, f.SpeciesName, f.Hyperlink, Date.Now.AddHours(-24 * rnd1.NextDouble())))
                    Next f
                End If
                Return data_Renamed.AsReadOnly()
            End Get
        End Property
        Private Shared Function GetDataSource() As List(Of Fish)
            Return DataSourceHelper.GetDataSouresFromXml(Of Fish)("fishes.xml", "Fishes")
        End Function
    End Class
End Namespace
